mnefun/_mnefun.py

```diff
@@ -77,6 +78,11 @@
 from mne.report import Report
 from mne.io.constants import FIFF
 
+try:
+    from autoreject import get_rejection_threshold
+except ImportError, e:
+    print(e)
```

Comment:
Nest import to `autoreject` in whatever nested clause it will actually be used. Then no need for `try/except`

Reply:
```
...
    elif func == do_preprocessing_combined:
        if p.autoreject_thresholds:
            from autoreject import get_rejection_threshold
        outs[ii] = func(p, subjects, run_indices)
....
```

For the sake of not DRY'ing why does above snippet not work in `do_processing`. I can see the code go into the conditional and executing ` if p.autoreject_thresholds` as `True` but I still get 

> NameError: global name 'get_rejection_threshold' is not defined

down in `do_processing_combined`

Reply:
It's a scope problem. When you import `get_rejection_threshold` in that function, the `func` (whatever it is, which must be `do_processing_combined`) does not have access to it. You should do the import in `do_processing_combined`.

Reply:
(A good introspecting editor will give you a flake error in both functions, saying that `get_rejection_threshold` isn't used in the first function, and isn't known in the second)

Reply:
Thanks. Pycharm has no native support for flake :disappointed: